module.json

```diff
@@ -20,6 +20,9 @@
   ],
   "dependencies": {},
   "targetDependencies": {
+    "st-ble-shield": {
+      "x-nucleo-idb0xa1": "ARMmbed/ble-x-nucleo-idb0xa1"
```

Comment:
This is also possible (it's a reference to a github repo, see here: http://yottadocs.mbed.com/reference/module.html#dependencies)

You can combine a version spec and source by using `user/Repo#~1.2.3` for example.